src/ZoneServer/Skills/Handlers/Sapper/Claymore.cs

```diff
@@ -0,0 +1,162 @@
+﻿using System;
+using System.Threading.Tasks;
+using Melia.Shared.L10N;
+using Melia.Shared.Tos.Const;
+using Melia.Shared.World;
+using Melia.Zone.Network;
+using Melia.Zone.Skills.Handlers.Base;
+using Melia.Zone.Skills.SplashAreas;
+using Melia.Zone.World.Actors;
+using Melia.Zone.World.Actors.CombatEntities.Components;
+using Melia.Zone.Skills.Combat;
+using Melia.Zone.World.Actors.Characters;
+using Melia.Zone.World.Actors.Monsters;
+using static Melia.Zone.Skills.SkillUseFunctions;
+using Melia.Shared.Data.Database;
+
+namespace Melia.Zone.Skills.Handlers.Sapper
+{
+	/// <summary>
+	/// Handler for the Sapper skill Claymore.
+	/// </summary>
+	[SkillHandler(SkillId.Sapper_Claymore)]
+	public class Claymore : IGroundSkillHandler
+	{
+		private Mob trap;
+
+		/// <summary>
+		/// Handles the skill, creates an trap object on the floor on
+		/// the first usage and explodes it on the second usage attempt
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		/// <param name="originPos"></param>
+		/// <param name="farPos"></param>
+		/// <param name="designatedTarget"></param>
+		public void Handle(Skill skill, ICombatEntity caster, Position originPos, Position farPos, ICombatEntity designatedTarget)
+		{
+			if (!caster.TrySpendSp(skill))
+			{
+				caster.ServerMessage(Localization.Get("Not enough SP."));
+				return;
+			}
+
+			if (!caster.Position.InRange2D(farPos, skill.Data.MaxRange))
+			{
+				caster.ServerMessage(Localization.Get("Too far away."));
+				return;
+			}
+
+			if (trap != null)
+			{
+				this.ExplodeTrap(caster, skill);
+				return;
+			}
+
+			this.PlaceTrap(caster, skill, farPos);
+		}
+
+		/// <summary>
+		/// Handles the trap triggering explosion effect
+		/// </summary>
+		/// <param name="caster"></param>
+		/// <param name="skill"></param>
+		private async void ExplodeTrap(ICombatEntity caster, Skill skill)
+		{
+			var character = caster as Character;
+
+			skill.IncreaseOverheat();
+			caster.SetAttackState(true);
+
+			await Task.Delay(200);
+
+			Send.ZC_NORMAL.Skill_88(character, character, null);
+			Send.ZC_NORMAL.Skill_88(character, caster, null);
+
+			Send.ZC_SKILL_READY(caster, null, caster.Position, caster.Position);
+			Send.ZC_NORMAL.UpdateSkillEffect(caster, 0, caster.Position, trap.Direction, trap.Position);
+			Send.ZC_SKILL_MELEE_GROUND(caster, skill, trap.Position, ForceId.GetNew(), null);
+
+			await Task.Delay(350);
+
+			Send.ZC_DEAD(trap, trap.Position);
+
+			caster.Map.RemoveMonster(trap);
```

Comment:
The trap wouldn't be removed if the caster changed maps or DCed in between casting and this happening.